streaming/constructor.go

```diff
@@ -0,0 +1,69 @@
+package streaming
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	serverTypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/streaming/file"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/spf13/cast"
+)
+
+// ServiceConstructor is used to construct a streaming service
+type ServiceConstructor func(opts serverTypes.AppOptions, keys []sdk.StoreKey, marshaller codec.BinaryMarshaler) (sdk.StreamingService, error)
+
+// ServiceType enum for specifying the type of StreamingService
+type ServiceType int
+
+const (
+	Unknown ServiceType = iota
+	File
+	// add more in the future
+)
+
+// NewStreamingServiceType returns the streaming.ServiceType corresponding to the provided name
+func NewStreamingServiceType(name string) ServiceType {
+	switch strings.ToLower(name) {
+	case "file", "f":
+		return File
+	default:
+		return Unknown
+	}
+}
```

Comment:
I think we may want to consider a plugin architecture with dynamic registration rather than everything defined up front. What do you think?

Reply:
We will be doing this in a separate PR: the work is currently here: https://github.com/i-norden/cosmos-sdk/pull/1